--- a/js/public/HashTable.h
+++ b/js/public/HashTable.h
@@ -550,14 +550,7 @@
         size_t word = reinterpret_cast<size_t>(l) >> zeroBits;
         static_assert(sizeof(HashNumber) == 4,
                       "subsequent code assumes a four-byte hash");
-#if JS_BITS_PER_WORD == 32
         return HashNumber(word);
-#else
-        static_assert(sizeof(word) == 8,
-                      "unexpected word size, new hashing strategy required to "
-                      "properly incorporate all bits");
-        return HashNumber((word >> 32) ^ word);
-#endif
     }
     static bool match(const Key& k, const Lookup& l) {
         MOZ_ASSERT(!JS::IsPoisonedPtr(k));
