--- a/js/src/jstypes.h
+++ b/js/src/jstypes.h
@@ -185,7 +185,7 @@
 #define JS_BITS_PER_BYTE 8
 #define JS_BITS_PER_BYTE_LOG2 3
 
-#if defined(JS_64BIT)
+#if defined(JS_64BIT) && ! defined(__ILP32__)
 # define JS_BITS_PER_WORD 64
 #else
 # define JS_BITS_PER_WORD 32

--- a/gfx/ycbcr/yuv_row_posix.cpp
+++ b/gfx/ycbcr/yuv_row_posix.cpp
@@ -9,7 +9,8 @@
 
 extern "C" {
 
-#if defined(ARCH_CPU_X86_64)
+// FIXME: implement x32 assembly (always SSE2)
+#if defined(ARCH_CPU_X86_64) && defined(__LP64__)
 
 // We don't need CPUID guards here, since x86-64 implies SSE2.

--- a/js/src/configure.in
+++ b/js/src/configure.in
@@ -2087,6 +2087,10 @@
     ENABLE_ION=1
     AC_DEFINE(JS_CPU_X86)
     ;;
+x86_64*-*x32)
+    dnl ENABLE_ION=1
+    dnl AC_DEFINE(JS_CPU_X32)
+    ;;
 x86_64*-*)
     ENABLE_ION=1
     AC_DEFINE(JS_CPU_X64)
